### PR TITLE
Add new Medium "Now Playing" widget

### PR DESCRIPTION
**Pull request by jgcaruso**
Follow up to https://github.com/Automattic/pocket-casts-ios/pull/1343 that introduces the new medium Now Playing widget from Lkt7fuf9Nq3XvfAFPCfpTD-fi-995_4119

Also adapts the existing "nothing playing" view for the medium size.

**iOS 17**
| light | dark |
| -- | -- |
| Simulator Screenshot - iPhone 15 Plus - 2024-07-04 at 14 54 43 | Simulator Screenshot - iPhone 15 Plus - 2024-07-04 at 14 55 18 |

**iOS 16**
| light | dark |
| -- | -- |
| Simulator Screenshot - iPhone SE (3rd generation) - 2024-07-04 at 15 42 59 | Simulator Screenshot - iPhone SE (3rd generation) - 2024-07-04 at 15 42 35 |

**Nothing Playing**
| light | dark |
| -- | -- |
| Simulator Screenshot - iPhone 15 Plus - 2024-08-01 at 16 23 14 | Simulator Screenshot - iPhone 15 Plus - 2024-08-01 at 16 23 00 |

## To test

1. pull branch, build and run on device or simulator
2. Open the widget chooser, scroll through the options for Pocket Casts. You should see the new medium Now Playing widget in both styles after the "small" variation
3. Add one (or both) to your home screen
4. The widget should match the designs
5. If you added the white version, flipping into "dark" mode should turn it black. Or if it was already black, flipping into "light" mode should turn it white.
6. In iOS 17, the widget should be interactive. Tapping play should start the podcast, tapping pause should stop it.
   - tapping anywhere else should open Pocket Casts to the episode
8. In iOS 16, tapping anywhere should open Pocket Casts to the episode
9. When there isn't a "Now Playing" episode, the default state will match the "small" size and simply show the current Pocket Casts icon -- Waiting on design to review this since it won't work well as-is if a custom icon is chosen p1720121313396209-slack-C05RR9P9RAT

NOTE: I'm not entirely sure when `showsWidgetContainerBackground` will ever be false for these widgets, but if you want to simulate that you can set `Line 8` of `NowPlayingEntryView.swift` force `var showsWidgetBackground = false` and then see how it renders. It won't be pretty because padding is removed for some reason, but that isn't done as part of this PR.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WidgetExtension/Now Playing/NowPlayingEntryView.swift [hunk 1]
```diff
@@ -18,39 +19,92 @@ struct NowPlayingWidgetEntryView: View {
...
-                .clearBackground()
-                .if(!showsWidgetBackground) { view in
-                    view
-                        .padding(.top)
-                        .padding(.bottom)
-                }
+            if family == .systemSmall {
+                smallWidget(playingEpisode: playingEpisode)
+            } else {
+                mediumWidget(playingEpisode: playingEpisode)
```
Comment (danielebogo) [inline WidgetExtension/Now Playing/NowPlayingEntryView.swift]: Nit:
I would use a `switch` here when dealing with enums

**Comment by danielebogo** [comment]
Following the test instructions, it worked as described 👌

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 - A new button is added under Account Settings: "Change Avatar". It opens the in-app Safari with Gravatar's Avatar Quick Editor. [#1770](https://github.com/Automattic/pocket-casts-ios/pull/1770)
 - Adds the ability to rate podcasts [#1879](https://github.com/Automattic/pocket-casts-ios/issues/1879)
 - Improve the sort by name algorithm [#1959](https://github.com/Automattic/pocket-casts-ios/issues/1959)
+- Adds a medium size for the Now Playing widget [#1876](https://github.com/Automattic/pocket-casts-ios/pull/1876)
```
Comment (danielebogo) [inline CHANGELOG.md]: Let's update this following the assigned milestone. For now I set it to 7.72

File: WidgetExtension/Now Playing/NowPlayingEntryView.swift [hunk 3]
```diff
@@ -18,39 +19,96 @@ struct NowPlayingWidgetEntryView: View {
...
-                    artwork(playingEpisode: playingEpisode)
+                    HStack(alignment: .bottom) {
+                        Text("Now Playing")
+                            .font(.caption)
+                            .textCase(.uppercase)
+                            .padding(topPadding)
+                            .foregroundColor(widgetColorScheme.bottomTextColor.opacity(0.6))
+                        Spacer()
+                        Image(widgetColorScheme.iconAssetName)
+                            .frame(width: 28, height: 28)
```
Comment (bjtitus) [inline WidgetExtension/Now Playing/NowPlayingEntryView.swift]: ⛏️  It seems like this icon size is used three times, could it be put into a Constants or some other local property?

**Review by bjtitus — APPROVED** [review]
👍 Testing looked good on both iOS 17 and iOS 16 with and without something playing.

Just left one small nit in the comments.

**Comment by danielebogo** [comment]
Pinging @david-gonzalez-a8c to give it some 👀 on the design

File: WidgetExtension/Now Playing/NowPlayingEntryView.swift (hunk 3 above)
Comment (jgcaruso) [reply WidgetExtension/Now Playing/NowPlayingEntryView.swift]: Good call. Wasn't sure where best to add it but I found a `CommonWidgetHelper` class to add the property to, and found all cases where widget icon frame was being defined (it was more than 3) and updated them all.

**Pull request merged**